### PR TITLE
LBFGSFunctionOptimizer

### PR DESCRIPTION
**Pull request by DirkToewe**
#### Description
L-BFGS optimization was a [requested feature](https://github.com/tensorflow/tfjs/issues/127). This PR offers an `LBFGSFunctionOptimizer` that allows the optimization for loss functions of type `Tensor1D => Scalar`. This is however still at least one step away from a training optimizer. Any suggestion as to how to wrap the `LBFGSFunctionOptimizer` into a `LBFGSOptimizer` for training is welcome.

---

##### For repository owners only:

Please remember to apply all applicable tags to your pull request.
Tags: FEATURE, BREAKING, BUG, PERF, DEV, DOC, SECURITY

For more info see: https://github.com/tensorflow/tfjs/blob/master/DEVELOPMENT.md

### CODE REVIEW

**Comment by DirkToewe** [comment]
Waiting for suggestions/requests/review. No more commits will be made.
